project/ScalazBuild.scala

```diff
@@ -4,13 +4,12 @@ import Keys._
 object Scalaz extends Build {
   val testDeps = Seq("org.scalacheck" %% "scalacheck" % "1.13.4" % "test")
 
-  def module(prjName: String) = Project(
-    id = prjName,
-    base = file(prjName)).settings(
+  def stdSettings(prjName: String) = Seq(
     name := s"scalaz-$prjName",
-    scalaVersion := "2.11.11",
+    scalaVersion := "2.12.2",
     scalacOptions ++= Seq("-feature","-deprecation", "-Xlint", "-language:higherKinds",
-                          "-Ydelambdafy:method", "-Ypartial-unification", "-target:jvm-1.8"),
+                          "-Ydelambdafy:method", "-Ypartial-unification", "-target:jvm-1.8", "-opt:l:project",
```

Comment:
It's definitely measurable. However, I can manually inline the critical section if necessary, which will achieve most of the gains. The optimizer will only inline methods so annotated (`@inline`), and it looks more carefully engineered than 2.11.x. 

Let me know how to handle it.